chronos/internal/asyncengine.nim

```diff
@@ -10,6 +10,92 @@
 
 {.push raises: [].}
 
+## This module implements the core asynchronos engine / dispatcher.
+##
+## The dispatcher acts as a kind of event loop. You must call ``poll`` on it
+## (or a function which does so for you such as ``waitFor`` or ``runForever``)
+## in order to poll for any outstanding events. The underlying implementation
```

Comment:
It might make sense first to establish what events are and where you're polling from.
The first paragraph regards the knowledge of them and that you can poll them as already established.

Reply:
rephrased, though the entry point is the guide rather than the module documentation - it explains the basics better